internal/bucket/minio/minio.go

```diff
@@ -159,23 +174,48 @@ func newCredsFromSecret(secret *corev1.Secret) *credentials.Credentials {
 }
 
 // newAWSCreds creates a new Minio credentials object for `aws` bucket provider.
-func newAWSCreds(bucket *sourcev1.Bucket, proxyURL *url.URL) *credentials.Credentials {
+//
+// This function is only called when Secret authentication is not available.
+//
+// Authentication priority order:
+// 1. Object Level Workload Identity (if serviceAccountName is specified)
+// 2. AWS credential chain (EC2 instance profiles, environment variables, etc.)
+func newAWSCreds(ctx context.Context, bucket *sourcev1.Bucket, o *options) (*credentials.Credentials, error) {
+	var opts auth.Options
+	opts.Apply(o.authOpts...)
+
+	// 1. Try Object Level Workload Identity if serviceAccountName is specified
+	if opts.ServiceAccount != nil {
```

Comment:
It is our intention to make `auth` handle also controller-level, it was designed to support both object-level and controller-level authentication across all Flux controllers. As part of implementing RFC-0010 we are actively making this migration, which started in Flux 2.6 for kustomize-controller and for the OCI APIs.